### ISSUE TITLE
Not Working: Firefox 46.0.x

### ISSUE THREAD

**Issue by Botlover** [diegocr/CleanLinks#126]
The addon is no longer working for me. I've tested it with Firefox/Cyberfox 46.0. It does, however, work in Waterfox 43.0, so it seems to only be an issue with recent versions. 

Let me know if you need other information. 

Thanks.

**Comment by atomGit** [comment]
i'm experiencing the same - not working in 46

FF 46
CL 2.7.1
Linux

example page: https://archive.is/Pa8hn  - see the "Read more" link near the bottom

**Comment by GitCurious** [comment]
I am also seeing the same thing (CyberFox v46.x)

**Comment by atomGit** [comment]
another related problem i'm seeing is problems with the toolbar button - when i dragged the button on to the toolbar CL began working, at least in some instances, though the list of cleaned links may or may not be present in the button context menu

**Comment by rhky** [comment]
UNSUBSCRIBE

**Comment by diegocr** [comment]
Hi guys, thanks for bringing that to my attention.

Do you see any error related to CL in the browser console? (you can open it using ctrl+shift+j)

**Comment by GitCurious** [comment]
I do not see any CL errors or warnings in the browser console.

The CL icon will stay grey (off) after being pressed once and does not come back on again when pressed again.
The current session list stays empty.
There is NO highlight style applied (when Event Delegation mode is off and Link Tracking is on)

I think the link is cleaned when testing on Google images (encrypted) - but I am not 100% certain about it.

I have removed, reset & reinstalled CL to test - Cyberfox x64 (v46.3)

I don`t know if any of this information is useful.

**Comment by atomGit** [comment]
i can confirm what @GitCurious says - cool nick by the way :)

the toolbar icon acts strange when enabling/disabling it and the context menu list may or may not display the cleaned links, even when cleaning has been done (evident by the highlight style)

also for the test link i provided (https://archive.is/Pa8hn), CL is now working with that page - it may have started working after i placed the CL icon on the toolbar (i didn't put it there after installing CL because i didn't want/need it)

**Comment by GitCurious** [comment]
I think I have discovered the reson for this strange behaviour....in my case at least.

I recently restored a backup Cyberfox profile in which I had previously *enabled* the e10s settings for testing reasons.

**I have now disabled the e10s settings and CL appears to be working again.**

**Comment by atomGit** [comment]
browser.tabs.remote.autostart is false in my case

**Comment by diegocr** [comment]
I have just remembered that someone else reported some issue when the toolbar icon/button is customized outside of the main browser toolbar, so certainly that could be one reason :-/

Additional rules of thumb are:
- Use Event Delegation mode (the DOM traversal one is outdated/obsolete, not further improved in a long time nor tested on newer Firefox versions)
- Electrolysis/e10s is not yet supported, hence CL won't work while it's enabled. 

If problems still persists, you can give a try the version 2.8a1 available here on Github, to install it you can use the [GitHubExtIns](https://addons.mozilla.org/addon/GitHub-Extension-Installer/) extension - fwiw, i'm using that version on Fx 46 with no issues at all.

Hope that helps, and thanks for the follow ups :+1:

**Comment by ldgbc** [comment]
Will e10s be supported?

**Comment by diegocr** [comment]
Yes, e10s will be supported, but i won't start working on it until its integration on the stable channel/release of Firefox is imminent.

**Comment by ng4ever** [comment]
diegocr could you please get Tree Style Tab and Clean Links to work together ? Thanks.

**Comment by jawz101** [comment]
Isn't e10s already the default in Fennec/FFox for Mobile since forever?

**Comment by diegocr** [comment]
It is? Dunno, I didn't played with Fennec for quite some time.

Apparently there seems to be several e10s-related bugs still open:

https://bugzilla.mozilla.org/show_bug.cgi?id=e10s

**Comment by J0WI** [comment]
Note that e10s is planned to ship with next release (19 July).

**Comment by diegocr** [comment]
Thanks for bringing that to my attention, although it isn't the first time I hear that :)

If it does really ships with the next release, expect e10s support in CL a few days later.

**Comment by J0WI** [comment]
I just wonder why you are going to break the Add-On for those days instead of already providing a new version that can be tested now.

**Comment by diegocr** [comment]
Time has shown that prematurely working on supporting a new/breaking feature could be a waste of time, because their implementation/specification could change at any time - i faced this multiple times, so I ended up only working on them when fully established (I.e. hitting the stable channel)
